modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -2221,4 +2221,48 @@ TEST(Calib3d_SolvePnP, inputShape)
     }
 }
 
+TEST(Calib3d_SolvePnPRansac, badInputPoints)
+{
+    // https://github.com/opencv/opencv/issues/17799
+    vector<Point3f> pts3ds;
+    vector<Point2f> pts2ds;
+    Mat rvec_est_iterative, tvec_est_iterative;
+    Mat rvec_est_epnp, tvec_est_epnp;
+    Mat camera_mat = Mat::zeros(3, 3, CV_32FC1);
+    camera_mat.at<float>(0, 0) = 741.341;
+    camera_mat.at<float>(0, 2) = 484.054;
+    camera_mat.at<float>(1, 1) = 741.341;
+    camera_mat.at<float>(1, 2) = 353.712;
+    camera_mat.at<float>(2, 2) = 1.f;
+
+    pts3ds = {
```

Comment:
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2247:5: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2238:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2238:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2238:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2258:5: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2249:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2249:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]
> /build/precommit_linux64/3.4/opencv/modules/calib3d/test/test_solvepnp_ransac.cpp:2249:12: warning: extended initializer lists only available with -std=c++11 or -std=gnu++11 [enabled by default]

See the buildbot (you can click on the build numbers): https://pullrequest.opencv.org/#/summary/opencv

This requires C++11 and 3.4 OpenCV cannot require C++11 features.

Reply:
Please help me what should I do to fix the build failures?

Reply:
Initializer list for `vector` is a C++11 feature which should not be used here. You can either `push_back` the points to the `vector`, or use `cv::Mat` instead. See my PR [here](https://github.com/mightbxg/opencv/commit/1b6cf69272a565ed4632c4d203ac3b355e1d2cb9#).